dev/src/generate/parser.rs

```diff
@@ -53,6 +53,8 @@ pub struct Config {
     pub value: ConfigType,
     /// If given config is optional or not.
     pub optional: bool,
+    /// if this field is deprecated, a deprecated message will be provided.
+    pub deprecated: Option<String>,
```

Comment:
Hi, I feel that generating a `deprecated` message for bindings is meaningless. 

The message is usually something like "please use `Abc` instead," which is not very helpful in the context of bindings. The message itself could be confusing.

Another issue is that the version specified in `core` is not the same as the one in the bindings.

Maybe we just need a `bool` here.

Reply:
> At least in python it would look like this, and as a python binding I think it's not meaningless, unless we have no plan to remove `batch_max_operations`:

It seems to make sense. The deprecation message for the config is usually meaningful as they are not related to types.

> Can you explain why "Another issue is that the version specified in core is not the same as the one in the bindings." ？ At least in python and nodejs bindings it's a local crate

Hi, let's take `batch_max_operations` as an example, the deprecate message will be:

```shell
Deprecated since 0.52.0: Please use delete_max_size instead of batch_max_operations
```

The `0.52.0` version refers to `core` rather than the binding itself. 

Our current implementation simply ignores the version field of the deprecated message, and it seems to work as well.

Perhaps we could also parse the version, allowing me to use it for Rust code generation?

Reply:
> We can also parse that version but that will cause problem of confusing version which we just talked about: the version is for core, not the binding.
> 
> Unless we match versions on all bindings with core it's meanless to parse this version.

Hi, I want to generate code for `core` so this version is exactly what I want.